packages/go_router/lib/src/delegate.dart

```diff
@@ -66,18 +102,33 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
       pageKey: pageKey,
     );
 
-    _matches.push(newPageKeyMatch);
+    _matchList.push(newPageKeyMatch);
     notifyListeners();
   }
 
-  /// Returns `true` if there is more than 1 page on the stack.
+  /// Returns `true` if the active Navigator can pop.
   bool canPop() {
-    return _matches.canPop();
+    // Loop through navigators in reverse and call canPop()
+    final int matchCount = _matchList.matches.length;
+    for (int i = matchCount - 1; i >= 0; i--) {
+      final RouteMatch match = _matchList.matches[i];
+      final RouteBase route = match.route;
+      if (route is ShellRoute) {
+        final NavigatorState? navigatorState = route.navigatorKey.currentState;
+        if (navigatorState != null) {
+          final bool canPopNavigator = navigatorState.canPop();
+          if (canPopNavigator) {
+            return true;
+          }
+        }
+      }
```

Comment:
same here this should check for GoRoute whose parentnavigatorkey != null

Reply:
Dont need to do it now, but we should probably not expose this... Do you know whats the use case?

Reply:
Same as above, I think we will pop from the root Navigator or ShellRoute Navigator

Reply:
This is fixed in d35a25b736d3b370953a11e10a7457fba8005b95